contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -1437,48 +1444,34 @@ export const Clipboard = new Resource({
       return this.table.bulkDelete(ids);
     });
   },
-  copy(node_id, channel_id, clipboardRootId, extra_fields = null) {
-    return this.transaction({ mode: 'rw' }, TABLE_NAMES.CONTENTNODE, () => {
-      return this.tableCopy(node_id, channel_id, clipboardRootId, extra_fields);
-    });
-  },
-
-  tableCopy(node_id, channel_id, clipboardRootId, extra_fields = null) {
-    const parent = clipboardRootId;
-    return this.table
-      .where({ parent })
-      .sortBy('lft')
-      .then(siblings => {
-        let lft = 1;
 
-        if (siblings.length) {
-          lft = siblings.slice(-1)[0].lft + 1;
-        }
+  copy(node_id, channel_id, clipboardRootId, extra_fields = null) {
+    return Promise.all([
+      ContentNode.get({ '[node_id+channel_id]': [node_id, channel_id] }),
+      this.where({ parent }).sortBy('lft'),
+    ]).then(([node, siblings]) => {
+      let lft = 1;
 
-        // Get source node so we can reference some specifics
-        const nodePromise = ContentNode.table.get({
-          '[node_id+channel_id]': [node_id, channel_id],
-        });
+      if (siblings.length) {
+        lft = siblings.slice(-1)[0].lft + 1;
+      }
 
-        // Next, we'll add the new node immediately
-        return nodePromise.then(node => {
-          const data = {
-            id: uuid4(),
-            lft,
-            source_channel_id: channel_id,
-            source_node_id: node_id,
-            root_id: clipboardRootId,
-            kind: node.kind,
-            parent,
-            extra_fields,
-          };
-          return this.table.put(data).then(() => ({
-            // Return the id along with the data for further processing
-            source_id: node.id,
-            ...data,
-          }));
-        });
+      // Next, we'll add the new node immediately
+      const data = {
+        id: uuid4(),
+        lft,
+        source_channel_id: channel_id,
+        source_node_id: node_id,
+        root_id: clipboardRootId,
+        kind: node.kind,
+        parent: clipboardRootId,
+        extra_fields,
+      };
+
+      return this.transaction({ mode: 'rw' }, TABLE_NAMES.CONTENTNODE, () => {
```

Comment:
Why do we need to add the content node table here? We've already fetched from the content node table above, this transaction is just dealing with a put to the clipboard table, no?

Reply:
I think this was a just a remnant of old code perhaps. Removed!